here-naksha-app-service/src/test/java/com/here/naksha/app/service/IterateFeaturesTest.java

```diff
@@ -170,4 +170,41 @@ void tc1103_testIterateWithInvalidHandle() throws URISyntaxException, Interrupte
             .hasJsonBody(expectedBodyPart, "Iterate Feature Error response body doesn't match");
   }
 
+  @Test
+  void tc1104_testIterateInTwoPagesWithPropSelection() throws URISyntaxException, InterruptedException, IOException {
+    // Test API : GET /hub/spaces/{spaceId}/iterate
+    // Validate all features getting returned in two iterations, with:
+    // - first iteration returning features and nextPageToken, selecting: 1 normal prop, 1 URI encoded prop, 1 prop not existing
+    // - second iteration accepting handle but use the wrong selection delimiter (should return 400)
+
+    // Given: iterate parameters for first request
+    final String limitQueryParam = "limit=3";
+    final String firstStreamId = UUID.randomUUID().toString();
+    final String firstExpectedBodyPart = loadFileOrFail("ReadFeatures/Iterate/TC1104_testIterateInTwoPagesWithPropSelection/iterate_response_1.json")
+            .replaceAll("\\{\\{streamId}}",firstStreamId);
+    final String selectionParams = "selection=p.speedLimit,p.unknown_prop,%s".formatted(urlEncoded("p.@ns:com:here:xyz.tags"));
```

Comment:
If the original Features have only speedLimit and tags, then from this Test, how can we assure that result only returned the properties as per "selection" param? I guess the test will pass even if you stop supporting property selection... is that true? If yes, then there is a need to have some additional attributes in original features.